src/Sema.zig

```diff
@@ -6507,9 +6507,19 @@ fn analyzeCall(
                     break :func_ty ptr_info.pointee_type;
                 }
             },
-            else => {},
+            else => |calleeTypeTag| {
+                const msg = try sema.errMsg(block, func_src, "type '{}' is not a function", .{callee_ty.fmt(sema.mod)});
+                errdefer msg.destroy(sema.gpa);
+                if (calleeTypeTag == .Optional) {
+                    var buf: Type.Payload.ElemType = undefined;
+                    if (callee_ty.optionalChild(&buf).zigTypeTag() == .Fn) {
+                        try sema.errNote(block, call_src, msg, "cannot call optional function type", .{});
```

Comment:
Now that this is next to the base error and reworded it's easy to see that I should've suggested to make this a proper error instead of a note.
```suggestion
                        const msg = try sema.errMsg(block, call_src, "cannot call optional function type", .{});
```